numba_dppy/tests/kernel_tests/test_return_type.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import numba_dppy as dppy
+import pytest
+import dpctl
+from numba_dppy.tests._helper import skip_test
+
+
+def f(a):
+    return a
+
+
+list_of_sig = [
+    None,
+    ("int32[::1](int32[::1])"),
+]
+
+
+@pytest.fixture(params=list_of_sig)
+def sig(request):
+    return request.param
+
+
+def test_return(offload_device, sig):
+    if skip_test(offload_device):
+        pytest.skip()
+
+    a = np.array(np.random.random(122), np.int32)
+
+    with pytest.raises(Exception):
+        kernel = dppy.kernel(sig)(f)
+
+        with dpctl.device_context(offload_device):
+            kernel[a.size, dppy.DEFAULT_LOCAL_SIZE](a)
```

Comment:
1. Exception should be concrete. Just catching any exception could hide a problem.
2. I think `dppy.kernel(sig)(f)` does not throw exception. So `with pytest.raises(Exception):` should wrap only function call.

Reply:
1. I agree but this function raises `TypeError` which is a generic error and is suseptible to your original comment. In this case should we make our own error type to make it actually concrete? Is this scalable?
2. It does and hence the wrap. If signature is provided the `exception` would be raised at the time this function is going through the decorator.